blocklylib-core/src/main/java/com/google/blockly/model/BlockTemplate.java

```diff
@@ -433,8 +432,6 @@ public String toString(String nounPrefix) {
      * @return This block template, for chaining.
      */
     BlockTemplate withFieldValue(String fieldName, String value) {
-        assert(!TextUtils.isEmpty(fieldName));
-        assert(!TextUtils.isEmpty(value));
```

Comment:
BlockFactory handles the situation where these are empty, so I'm removing these assertions.

Reply:
I'd prefer to keep it to highlight the error as close to the user's code as possible, rather than some later error.

Reply:
The issue is there are tests explicitly checking that we don't throw here. If I fix these assertions, then those tests start failing. Either the tests are wrong, or we shouldn't be throwing here.

Reply:
Okay.. Fixed in #704.  The field value can be empty, such as an empty text block.  You can leave this out of your commit.